src/AppContainer.js

```diff
@@ -1,22 +1,54 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { Route, Switch } from "react-router-dom";
 import { hot } from "react-hot-loader/root";
+import { CSSTransition } from "react-transition-group";
 import { routes } from "./routes";
 import { NavigationBar } from "./components/Layout/NavigationBar";
 import { FooterDefault } from "./components/Layout/Footer";
+import { Overlay } from "./components/UI/Overlay";
+import { useKeyPress } from "./common/hooks/useKeyPress";
+import { OverlayProvider } from "./common/contexts/OverlayContext";
 
 const AppContainer = () => {
+  const [overlayContent, setOverlayContent] = useState(<></>);
```

Comment:
can get rid of setOverlayVisible

=> overlayContent is set to null when should be hidden